src/lib/expansion/expansion-panel.ts

```diff
@@ -26,17 +27,43 @@ import {
   SkipSelf,
   ViewContainerRef,
   ViewEncapsulation,
+  Inject,
 } from '@angular/core';
 import {Subject} from 'rxjs';
 import {filter, startWith, take} from 'rxjs/operators';
-import {MatAccordion} from './accordion';
 import {matExpansionAnimations} from './expansion-animations';
 import {MatExpansionPanelContent} from './expansion-panel-content';
 
+/** MatAccordion's display modes. */
+export type MatAccordionDisplayMode = 'default' | 'flat';
 
 /** MatExpansionPanel's states. */
 export type MatExpansionPanelState = 'expanded' | 'collapsed';
 
+/**
+ * Base interface for a `MatAccordion`.
+ * @docs-private
+ */
+export interface MatAccordionBase extends CdkAccordion {
```

Comment:
Shouldn't `MatAccordionDisplayMode`, `MatAccordionBase` and `MAT_ACCORDION` be defined in the `accordion.ts` file rather than the `expansion-panel.ts` file?

Reply:
They should, but then we'd get a circular import because `accordion.ts` imports `expansion-panel.ts`.

Reply:
Can we moved it to another file then?

I think the discovery of the type and interface doesn't make sense since they are for accordion.  We could just have an `accordion-base.ts` file?

Reply:
Done.